File: Python_Library/Film.py
from Media import Media

class Film(Media):
	def __init__(self,a,b,c,d,e,f):
		Media.__init__(self,a,b,c,e)
		self.director=d
		self.year=f
		
		
	def display(self):
		print("<------------------------------ Record From Film --------------------------------------------->\n");
		print("Call Number: "+self.call_number+"\n");
		print("Title: "+self.title+"\n");
		print("Subjects: "+self.subject+"\n");
		print("Director: "+self.director+"\n");
		print("Notes: "+self.notes+"\n");
		print("Year: "+self.year+"\n");
		
	def compare_other(self,x):
		if (self.director.find(x) != -1):
			return True
		elif(self.year.find(x) !=-1):
			return True
		elif(self.notes.find(x) !=-1):
			return True
		else:
			return False
	
 
  

File: ObjectOrientedDesign(LibrarySearch)/Film.cpp
#include<iostream>
#include<string>
#include"Film.h"
#include"Media.h"
using namespace std;


Film :: Film( string a,string b,string c,string d, string e, string f) : Media(a, b, c, e)
{  
  this->director=d;  
  this->year=f; 
  
}

void Film::display() const
{
   cout<<"<------------------------------ Record From Film--------------------------------------------->"<<endl;
   cout<<"Call Number:"<<call_number<<"\n"<<"Title:"<<title<<"\n"<<"Subjects:"<<subject<<"\n"<<"Director:"<<director<<"\n"<<"Notes:"<<notes<<
  "\n"<<"Year:"<<year<<endl;
}


bool Film::compare_other(const string& x) const
{
  
  if (this->notes.find(x) != std::string::npos)
  {
     return true;
  } 
  else if(this->director.find(x) != std::string::npos)
   {
     return true;
   }
    else if(this->year.find(x) != std::string::npos) 
   {
     return true;
   }
   else 
   {
     return false;
   }
}

